packages/web-components/src/components/ic-data-table/ic-data-table.tsx

```diff
@@ -0,0 +1,493 @@
+import { Component, Element, h, Prop, State, Listen } from "@stencil/core";
+import unsortedIcon from "./assets/unsorted-icon.svg";
+import ascendingIcon from "./assets/ascending-icon.svg";
+import descendingIcon from "./assets/descending-icon.svg";
+import {
+  IcDataTableColumnObject,
+  IcDataTableDensityOptions,
+  IcDataTableSortOrderOptions,
+} from "./ic-data-table.types";
+import {
+  IcPaginationAlignmentOptions,
+  IcPaginationControlTypes,
+  IcPaginationTypes,
+} from "../ic-pagination/ic-pagination.types";
+
+@Component({
+  tag: "ic-data-table",
+  styleUrl: "ic-data-table.css",
+  shadow: true,
+})
+export class DataTable {
+  private SORT_ICONS = {
+    unsorted: unsortedIcon,
+    ascending: ascendingIcon,
+    descending: descendingIcon,
+  };
+
+  @Element() el: HTMLIcDataTableElement;
+
+  @State() fromRow: number = 0;
+
+  @State() previousRowsPerPage: number;
+
+  @State() rowsPerPage: number;
+
+  @State() scrollable: boolean = false;
+
+  @State() scrollOffset: number = 0;
+
+  @State() selectedRow: object;
+
+  @State() sortedColumn: string;
+
+  @State() sortedColumnOrder: IcDataTableSortOrderOptions;
+
+  @State() toRow: number;
+
+  /**
+   * The title for the table only visible to screen readers.
+   */
+  @Prop() caption!: string;
+
+  /**
+   * The column headers for the table.
+   */
+  @Prop() columns!: IcDataTableColumnObject[];
+
+  /**
+   * The row content for the table.
+   */
+  @Prop() data: { [key: string]: any }[];
+
+  /**
+   * Set the density of the table including font and padding.
+   */
+  @Prop() density?: IcDataTableDensityOptions = "default";
```

Comment:
Might be something to discuss with design, but does this need to be renamed to size? It seems a bit inconsistent as we don't have a density prop on any other components, and the size prop tends to only affect the padding/font too. We could also then use the IcSizes type as it would need to be small, default, large instead of dense, default, spacious.

Reply:
Have previously discussed this with design. This has deemed to be more reflective of the actual changes being applied, whereas 'sizes' is quite misleading